core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -1644,13 +1677,52 @@ class KafkaZkClient private (zooKeeperClient: ZooKeeperClient, isSecure: Boolean
       }
     }
 
+    // This method is part of the work around done in the KAFKA-7165, once ZOOKEEPER-2985 is complete, this code must
+    // be deleted, if by the time you are reading this the code is still present and the bug solve, please delete it.
+    private def delete(): Code = {
+      val deleteRequest = DeleteRequest(path, ZkVersion.MatchAnyVersion)
+      val deleteResponse = retryRequestUntilConnected(deleteRequest)
+      deleteResponse.resultCode match {
+        case code@ Code.OK => code
+        case code =>
+          error(s"Error while deleting ephemeral node at $path with return code: $code")
+          code
+      }
+    }
+
+    private def reCreate(): Code = {
+      val codeAfterDelete = delete()
+      var codeAfterReCreate = codeAfterDelete
+      info(s"Result of znode ephemeral deletion at $path is: $codeAfterDelete")
+      if (codeAfterDelete == Code.OK) {
+        codeAfterReCreate = create()
+        info(s"Result of znode ephemeral re-creation at $path is: $codeAfterReCreate")
+      }
+      codeAfterReCreate
+    }
+
     private def getAfterNodeExists(): Code = {
       val getDataRequest = GetDataRequest(path)
       val getDataResponse = retryRequestUntilConnected(getDataRequest)
+      val ephemeralOwnerId = getDataResponse.stat.getEphemeralOwner
       getDataResponse.resultCode match {
-        case Code.OK if getDataResponse.stat.getEphemeralOwner != zooKeeperClient.sessionId =>
+        // At this point, the Zookeeper session could be different (due a 'Session expired') from the one that initially
+        // registered the Broker into the Zookeeper ephemeral node, but the znode is still present in ZooKeeper.
+        // The expected behaviour is that Zookeeper removes the ephemeral node associated with the expired session but
+        // due an already reported bug in Zookeeper (ZOOKEEPER-2985) this is not happening, so, the following check will
+        // validate if this Broker got registered with the previous (expired) session and try to register again, deleting
+        // the ephemeral node and creating it again.
+        // This code is part of the work around done in the KAFKA-7165, once ZOOKEEPER-2985 is complete, this code must
+        // be deleted, if by the time you are reading this the code is still present and the bug solve, please delete it.
+        case Code.OK if shouldReCreateEphemeralZNode(ephemeralOwnerId) =>
+          warn(s"Error while creating ephemeral at $path, node already exists and owner " +
```

Comment:
Should this be info instead of warn? We probably want to void the word Error in an info logging.